pkg/util/common/common.go

```diff
@@ -39,36 +39,45 @@ func (s StringSet) GetAll() []string {
 // StructToMap converts a struct to a map[string]interface{} based on `json` annotations defaulting to field names
 func StructToMap(obj interface{}) map[string]interface{} {
 	rt, rv := reflect.TypeOf(obj), reflect.ValueOf(obj)
-	if rt != nil && rt.Kind() != reflect.Struct {
-		return make(map[string]interface{}, 0)
+
+	out := make(map[string]interface{}, 0)
+
+	if rt == nil || rv.IsZero() {
+		return out
 	}
 
-	out := make(map[string]interface{}, rt.NumField())
-	for i := 0; i < rt.NumField(); i++ {
-		field := rt.Field(i)
+	if rt.Kind() == reflect.Ptr {
+		obj = rv.Elem().Interface()
+		rt, rv = reflect.TypeOf(obj), reflect.ValueOf(obj)
```

Comment:
Did you want to recurse here, to handle the `rt == nil` and `rt.Kind() == reflect.Ptr` cases?